### PR TITLE
DPL: add type trait helper for compile-time strings

### PR DESCRIPTION
**Pull request by mario-krueger**

### CODE REVIEW

File: Framework/Core/test/test_StringHelpers.cxx [hunk 1]
```diff
@@ -42,4 +44,13 @@ BOOST_AUTO_TEST_CASE(StringHelpersConstStr)
   static_assert(std::is_same_v<decltype(myConstStr), ConstStr<'h', 'e', 'l', 'l', 'o', 'W', 'o', 'r', 'l', 'd'>>);
   static_assert(myConstStr.hash == (uint32_t)942280617);
   BOOST_CHECK_EQUAL(myConstStr.hash, compile_time_hash("helloWorld"));
+
+  if constexpr (is_const_str<decltype(myConstStr)>::value) {
```
Comment (ktf) [inline Framework/Core/test/test_StringHelpers.cxx]: why not making it a constexpr function, rather than having it act on the type?

Comment (ktf) [reply Framework/Core/test/test_StringHelpers.cxx]: I mean something like:

```c++
template <typename T>
constexpr bool is_const_str_v(T) {
  return is_const_str<T>::value;
}
```

Comment (mario-krueger) [reply Framework/Core/test/test_StringHelpers.cxx]: great idea, thanks! I will add it.

**Comment by ktf** [comment]
@TimoWilken this is still failing. Could you take a look?

**Review by ktf — APPROVED** [review]

**Comment by TimoWilken** [comment]
@ktf It's an error building TpcFecUtils -- I've enabled debugging mode, more to follow.

**Pull request merged**